### PR TITLE
Add non-goals section documenting should vs expect

### PR DESCRIPTION
**Pull request by sauloperez**
This will hopefully reduce the number of issues asking for non-goals of RuboCop RSpec. Addresses what was suggested in https://github.com/rubocop-rspec/rubocop-rspec/issues/587#issuecomment-376343881

### CODE REVIEW

**Comment by sauloperez** [comment]
@dgollahon any feedback?

**Comment by bquorning** [comment]
Just to be clear: I pinged @dgollahon because he is a native English speaker (I assume).

File: README.md [hunk 1]
```diff
@@ -94,6 +94,28 @@ RSpec/FilePath:
     - spec/my_poorly_named_spec_file.rb
 ```
 
+## Non-goals of RuboCop RSpec
+
+### Enforcing `should` vs. `expect` syntax
+
+Enforcing:
```
Comment (dgollahon) [inline README.md]: If you're going to make this read like a sentence, i suggest dropping this `:` as well as the one after "over" and making "Is" on the following line lowercase.

File: README.md [hunk 2]
```diff
@@ -94,6 +94,28 @@ RSpec/FilePath:
...
+expect(calculator.compute(line_item)).to eq(5)
+```
+
+over:
+
+```ruby
+calculator.compute(line_item).should == 5
+```
+
+Is a feature of RSpec itself – here you can read about how to enable/disable one or the other: https://relishapp.com/rspec/rspec-expectations/docs/syntax-configuration#disable-should-syntax
```
Comment (dgollahon) [inline README.md]: How about using a markdown link here?

You could write this as

> is a feature of RSpec itself – you can read about it in the [RSpec Documentation](https://relishapp.com/rspec/rspec-expectations/docs/syntax-configuration#disable-should-syntax).

File: README.md [hunk 3]
```diff
@@ -94,6 +94,28 @@ RSpec/FilePath:
...
+
+Is preferable to configure RSpec to not do any monkey patching at all (no `should` method on every object, and the outer-most `describe` or `context` must be called directly on RSpec): https://relishapp.com/rspec/rspec-core/v/3-7/docs/configuration/zero-monkey-patching-mode#monkey-patched-methods-are-undefined-with-%60disable-monkey-patching!%60
+
+Before disabling `should` you will need all your specs to use the `expect` syntax for them to work. You can use [Transpec](http://yujinakayama.me/transpec/), which will do the conversion for you.
```
Comment (dgollahon) [inline README.md]: If you take my subheading suggestion from above, i think this should go into the previous section. Thanks for thinking to include this (I was mentally preparing to comment that we should mention Transpec but you were way ahead of me).

nit: "for them to work" is probably not necessary to communicate your point. fine to leave it if you prefer though.

**Comment by dgollahon** [comment]
> because he is a native English speaker (I assume)

Yup! 🇺🇸

**Comment by dgollahon** [comment]
@sauloperez If you're inclined to go the extra mile, it wouldn't hurt to add an extra bullet point to the [contributing guidelines](https://github.com/rubocop-rspec/rubocop-rspec/blob/master/.github/CONTRIBUTING.md#issue-reporting) saying they should check if the feature is a non-goal of `rubocop-rspec`.

Not a requirement though :)

**Comment by bquorning** [comment]
@sauloperez Sorry, but you need to rebase as well, to get the last (required) CircleCI job running. It was added to master after you opened this PR.

**Comment by sauloperez** [comment]
cool guys. On it right now.

**Comment by dgollahon** [comment]
🎉

**Pull request merged**